design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,176 @@
+Aligning the deployment process with K8s standards will particularly benefit use cases involving central control management, such as Multi-Cloud Computing (MCC) or Hybrid Cloud Mesh (HCM).
+
+### Prior discussion and links
+
+Not applicable.
+
+## User/User Story
+
+This is an example to user guide with all the details to install the ClusterLink project on a K8s Cluster.
+
+### Get ClusterLink project
+Clone the ClusterLink project:
+
+    git clone git@github.com:clusterlink-net/clusterlink.git
+
+Install the ClusterLink CLI:
+
+    cd clusterlink
+    make install
+
+In the future, these two commands should be replaced the command:
+
+    curl -L https://clusterlink.net/install.sh | sh
+
+### Create Fabric certificate
+Create the fabric certificate, which is the root CA certificate for all the K8s clusters:
+
+    clusterlink create fabric --name <fabric_name>
+
+The root ca certificates will be saved in the directory ```~/.config/clustrlink/<fabric_name>```.
+
+### Install ClusterLink on K8s Cluster
+The easiest way to install ClusterLink on the K8s cluster is by using the command:
+
+    clusterlink install --peer <name>
+
+This command installs the ClusterLink operator and creates the ClusterLink deployment.  
+Alternatively you can do it in steps:
+First, create and deploy the peer certificate:
+
+    clusterlink create peer --name <name>
+    kubectl create -f ```~/.config/clustrlink/<fabric_name>/<peer_nam>/secret.yaml```
+
+Second, create the ClustreLink operator:
+
+    kubectl create -f config/setup/
+
+Then, create clusterlink deployment:
+```
+kubectl apply -f - <<EOF
+apiVersion: clusterlink.net/v1alpha1
+kind: ClusterLink
+metadata:
+  namespace: clusterlink-ns
+  name: peer1
+spec:
+  dataplaneType: "envoy"
+  dataplaneReplicates: 1
+EOF
+```
+The operator will create:
+* ClusterLink controlplane deployment, including controlplane-pod, controlplane-service, and RBAC roles.  
+* ClusterLink dataplane deployment, including dataplane-pod and dataplane-service.
+* ClusterLink external access point using load-balancer/node-port/gateway-API service.
+  
+Overall, the ClusterLink deployment stages are:
+
+<img src="deployment.png" width="800" height="400" alt="ClusterLink Deployment Stages"/>    
+
+
+
+## Goals
+This design document should:
+
+- Define the steps for deploying a ClusterLink gateway to a K8s cluster.
+- Define the task functions of the ClusterLink operator.
+- Define the CRD (Custom Resource Definition) for the ClusterLink operator.
+
+## Non-Goals
+The following aspects are explicitly excluded and are out of scope in the current design:
+
+- This document focuses only on the deployment of ClusterLink to the K8s cluster. 
+The deployment to another environment as VMs is out of scope. 
+- The security aspects and user permissions concerning the creation of certificates, specifically for fabric or peer.
+## Proposal
```

Comment:
we should start running YAML linting. While it is displayed fine, there are expectations around use of empty lines between sections, etc.